openedx/core/djangoapps/credit/services.py

```diff
@@ -0,0 +1,130 @@
+            NONE (user not found or is not enrolled or is not credit course)
+            - or -
+            {
+                'enrollment_mode': the mode that the user is enrolled in the course
+                'profile_fullname': the name that the student registered under, used for verification
+                'credit_requirement_status': the user's status in fulfilling those requirements
+            }
+        """
+
+        # This seems to need to be here otherwise we get
+        # circular references when starting up the app
+        from openedx.core.djangoapps.credit.api.eligibility import (
+            is_credit_course,
+            get_credit_requirement_status,
+        )
+
+        # course_key = CourseKey.from_string(course_id)
+
+        try:
+            user = User.objects.get(id=user_id)
+        except ObjectDoesNotExist:
+            # bad user_id
+            return None
+
+        # since we have to do name matching during various
+        # verifications, User must have a UserProfile
+        try:
+            profile = UserProfile.objects.get(user_id=user_id)
+        except ObjectDoesNotExist:
+            # this shouldn't happen under normal circumstances
+            return None
+
+        course_key = (
+            CourseKey.from_string(course_key_or_id)
+            if isinstance(course_key_or_id, basestring)
+            else course_key_or_id
+        )
+
+        enrollment = CourseEnrollment.get_enrollment(user, course_key)
+        if not enrollment or not enrollment.is_active:
+            # not enrolled
+            return None
+
+        if not is_credit_course(course_key):
+            return None
+
+        return {
+            'enrollment_mode': enrollment.mode,
+            'profile_fullname': profile.name,
+            'credit_requirement_status': get_credit_requirement_status(course_key, user.username)
+        }
+
+    def set_credit_requirement_status(self, user_id, course_key_or_id, req_namespace,
+                                      req_name, status="satisfied", reason=None):
+        """
+        A simple wrapper around the method of the same name in api.eligibility.py. The only difference is
+        that a user_id is passed in.
+
+        For more information, see documentation on this method name in api.eligibility.py
+        """
+
+        log_msg = (
+            'set_credit_requirement_status was called with '
+            'user_id={user_id}, course_key_or_id={course_key_or_id} '
+            'req_namespace={req_namespace}, req_name={req_name}, '
+            'status={status}, reason={reason}'.format(
+                user_id=user_id,
+                course_key_or_id=course_key_or_id,
+                req_namespace=req_namespace,
+                req_name=req_name,
+                status=status,
+                reason=reason
+            )
+        )
+        log.info(log_msg)
+
+        # need to get user_name
+        user = User.objects.get(id=user_id)
+
+        course_key = (
```

Comment:
I've seen this small code block a couple times -- seems like a good candidate for a tiny helper method